### PR TITLE
Hypershift: Use token minter instead of a kubeconfig in ovn-kubernetes master

### PR DESCRIPTION
**Pull request by kyrtapz**

### CODE REVIEW

File: pkg/network/ovn_kubernetes.go [hunk 1]
```diff
@@ -83,18 +83,19 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
...
 	data.Data["OvnImage"] = os.Getenv("OVN_IMAGE")
 	data.Data["KubeRBACProxyImage"] = os.Getenv("KUBE_RBAC_PROXY_IMAGE")
-	data.Data["KUBERNETES_SERVICE_HOST"] = h
-	data.Data["KUBERNETES_SERVICE_PORT"] = p
-	data.Data["K8S_APISERVER"] = fmt.Sprintf("https://%s:%s", h, p)
+	data.Data["KUBERNETES_SERVICE_HOST"] = apiServer.Host
+	data.Data["KUBERNETES_SERVICE_PORT"] = apiServer.Port
+	data.Data["K8S_APISERVER"] = fmt.Sprintf("https://%s:%s", apiServer.Host, apiServer.Port)
+	data.Data["K8S_LOCAL_APISERVER"] = fmt.Sprintf("https://%s:%s", localAPIServer.Host, localAPIServer.Port)
+	data.Data["TokenAudience"] = os.Getenv("TOKEN_AUDIENCE")
```
Comment (squeed) [inline pkg/network/ovn_kubernetes.go:98]: Add a comment explaining vaguely what this is :)

**Comment by squeed** [comment]
/approve
Once you fix the image, we can lgtm it

File: bindata/network/ovn-kubernetes/managed/ovnkube-master.yaml [hunk 2]
```diff
@@ -63,7 +63,25 @@ spec:
                 app: ovnkube-master
             topologyKey: topology.kubernetes.io/zone
       priorityClassName: "system-cluster-critical"
+      # don't mount service account token as this is going to be generated by the token-minter container
+      automountServiceAccountToken: false
       containers:
+      # token-minter creates a token with the default service account path
+      # The token is read by ovn-k containers to authenticate against the hosted cluster api server
+      - name: token-minter
+        image: "quay.io/pdiak/hypershift:dev"
```
[the remark this answers is not in the document]
Comment (kyrtapz) [reply bindata/network/ovn-kubernetes/managed/ovnkube-master.yaml:72]: Wow, Of course!

**Comment by kyrtapz** [comment]
token-minter usage changed, updated it to reflect that

**Pull request merged**